apps/odyssey-stats/src/components/odyssey-query-site-purchases.ts

```diff
@@ -8,47 +8,71 @@ import { useEffect } from 'react';
 import { useDispatch } from 'react-redux';
 import wpcom from 'calypso/lib/wp';
 import getDefaultQueryParams from 'calypso/my-sites/stats/hooks/default-query-params';
+import { useSelector } from 'calypso/state';
 import {
 	PURCHASES_SITE_FETCH,
 	PURCHASES_SITE_FETCH_COMPLETED,
 	PURCHASES_SITE_FETCH_FAILED,
 } from 'calypso/state/action-types';
+import getEnvStatsFeatureSupportChecks from 'calypso/state/sites/selectors/get-env-stats-feature-supports';
 import { getApiNamespace, getApiPath } from '../lib/get-api';
 
-async function queryOdysseyQuerySitePurchases( siteId: number | null ) {
+async function queryOdysseyQuerySitePurchases(
+	siteId: number | null,
+	shouldUseStatsBuiltInPurchasesApi: boolean
+) {
 	if ( ! siteId ) {
 		return;
 	}
 
-	return wpcom.req
-		.get( {
-			path: getApiPath( '/site/purchases', { siteId } ),
-			apiNamespace: getApiNamespace(),
-		} )
-		.then( ( res: { data: string } ) => JSON.parse( res.data ) )
-		.catch( ( error: APIError ) => error );
+	const apiPath = shouldUseStatsBuiltInPurchasesApi
+		? `/sites/${ siteId }/purchases`
+		: '/site/purchases';
+
+	return (
+		wpcom.req
+			.get( {
+				path: getApiPath( apiPath, { siteId } ),
+				apiNamespace: getApiNamespace(),
```

Comment:
```suggestion
				apiNamespace: getApiNamespace( 'jetpack/v4/stats-app' ),
```
The [namespace](https://github.com/Automattic/jetpack/pull/38150/files#diff-8cc69be5879a24c41f547c9b12ecfaae58566d6d4281df42e4ee12f475ffec15R409) of the introduced Jetpack purchases API endpoint seems to be `/jetpack/v4/stats-app`.

Reply:
Good catch! fixed in aa258b9f29d